### PR TITLE
[ LSTM ] Optimize LSTM Gradient calculation

### PR DESCRIPTION
**Pull request by jijoongmoon**
Gradient computaion of LSTM takes over 60% of computation. This patch
includes the optimization using the tensor dimenson which only has
width.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: nntrainer/layers/lstmcell_core.cpp [hunk 1]
```diff
@@ -125,8 +125,40 @@ void lstmcell_calcGradient(
...
+    for (unsigned int i = 0; i < d_weight_hh.height(); ++i) {
+      unsigned int out_width = d_weight_hh.width();
+      float in_hh = prev_hidden_state.getValue(i);
+
+      float *d_weight_hh_address = d_weight_hh.getAddress(i * out_width);
+
+      float *d_ifgo_address = d_ifgo.getData();
+
+      for (unsigned int j = 0; j < out_width; ++j) {
+        d_weight_hh_address[j] += d_ifgo_address[j] * in_hh;
```
Comment (DonghakPark) [inline nntrainer/layers/lstmcell_core.cpp:157]: wouldn't it be possible to optimize performance more by using ```loop rolling```?

**Review by DonghakPark — APPROVED** [review]
LGTM

File: nntrainer/layers/lstmcell_core.cpp (hunk 1 above)
Comment (jijoongmoon) [reply nntrainer/layers/lstmcell_core.cpp:167]: I change it to use BLAS Lib. Thanks.

**Review by songgot — APPROVED** [review]
LGTM

**Pull request merged**